lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -617,29 +666,24 @@ export class UserAgentApplication {
    * @returns {Promise.<string>} - A Promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the token or error.
    */
   // TODO: params to accept AuthRequest object instead
-  acquireTokenPopup(scopes: Array<string>): Promise<string>;
-  acquireTokenPopup(scopes: Array<string>, authority: string): Promise<string>;
-  acquireTokenPopup(scopes: Array<string>, authority: string, user: User): Promise<string>;
-  acquireTokenPopup(scopes: Array<string>, authority: string, user: User, extraQueryParameters: string): Promise<string>;
-  acquireTokenPopup(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
+  acquireTokenPopup(request: AuthenticationParameters): Promise<string> {
     return new Promise<string>((resolve, reject) => {
       // Validate scopes
       // TODO: Replace with new validation pattern
-      const isValidScope = this.validateInputScope(scopes);
+      const isValidScope = this.validateInputScope(request.scopes);
       if (isValidScope && !Utils.isEmpty(isValidScope)) {
         // TODO: Should reject with custom error
         // TODO: Is this always accessToken?
         reject(ErrorCodes.inputScopesError + Constants.resourceDelimiter + isValidScope);
       }
 
-      if (scopes) {
-        scopes = this.filterScopes(scopes);
+      if (request.scopes) {
+        request.scopes = this.filterScopes(request.scopes);
       }
+      const scope = request.scopes.join(" ").toLowerCase();
 
-      const scope = scopes.join(" ").toLowerCase();
-
-      // Get the user object if session exists
-      const userObject = user ? user : this.getUser();
+      // Get the user object if a session exists
+      const userObject = request.account ? request.account : this.getUser();
```

Comment:
request.account || this.getUser();

Reply:
Left this alone. if this has to be changed, there are multiple places in the code where we are using the ternary operator. I would rather do this later in a separate PR once we move the critical stuff out.